modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/processors/SW360Enricher.java

```diff
@@ -82,13 +83,170 @@ public void configure(Map<String, String> configMap) throws AntennaConfiguration
+        if (sw360Release.getExternalIds().containsKey(SW360Attributes.RELEASE_EXTERNAL_ID_DLICENSES)) {
+            Optional<LicenseInformation> licenseInformation = artifact.askForGet(DeclaredLicenseInformation.class);
+
+            List<SW360SparseLicense> sw360SparseLicenses =
+                    getSW360SparseLicensesFromString(sw360Release.getExternalIds().get(SW360Attributes.RELEASE_EXTERNAL_ID_DLICENSES), ";");
+
+            List<License> artifactDeclaredLicenses = getLicenseListWithSW360AndArtifactLicenses(sw360SparseLicenses, licenseInformation);
+
+            artifact.addFact(new DeclaredLicenseInformation(LicenseSupport.mapLicenses(
+                    artifactDeclaredLicenses.stream()
+                            .map(License::getName)
+                            .collect(Collectors.toList())
+            )));
+        }
+    }
+
+    private List<License> getLicenseListWithSW360AndArtifactLicenses(List<SW360SparseLicense> sw360SparseLicenses, Optional<LicenseInformation> licenseInformation) {
+        List<License> artifactLicenses;
+        if (licenseInformation.isPresent()) {
+            artifactLicenses = licenseInformation.get().getLicenses();
+        } else {
+            artifactLicenses = new ArrayList<>();
+        }
+
+        if (hasDifferentLicenses(artifactLicenses, sw360SparseLicenses)) {
+            for (SW360SparseLicense sw360SparseLicense : sw360SparseLicenses) {
+                License tempLicense = new License();
+                tempLicense.setName(sw360SparseLicense.getFullName());
+                License license = makeLicenseFromLicenseDetails(new SW360License(tempLicense));
+                artifactLicenses.add(license);
+            }
+        }
+        return artifactLicenses;
+    }
+
+    private List<SW360SparseLicense> getSW360SparseLicensesFromString(String licenseString, String splitRegex) {
+        String[] licenses = licenseString.split(splitRegex);
+        List<SW360SparseLicense> sw360SparseLicenses = new ArrayList<>();
+
+        for (String license : licenses) {
+            SW360SparseLicense sw360SparseLicense = new SW360SparseLicense();
+            sw360SparseLicenses.add(sw360SparseLicense.setFullName(license));
+        }
+
+        return sw360SparseLicenses;
+    }
+
+    private void mapCoordinates(SW360Release sw360Release, Artifact artifact) {
+        Set<String> keySet = sw360Release.getExternalIds().keySet();
+
+        for (Class<? extends ArtifactCoordinates> key : SW360CoordinateKeysToArtifactCoordinates.getKeys()) {
+            String coordinateType = SW360CoordinateKeysToArtifactCoordinates.get(key);
+            if (keySet.contains(coordinateType)) {
+                String coordinateValue = sw360Release.getExternalIds().get(coordinateType);
+                String[] splitCoordiantes = coordinateValue.split(":");
+
+                ArtifactCoordinates coordinates;
+                if(splitCoordiantes.length == 3){
```

Comment:
inconsistent formatting:
```suggestion
                if (splitCoordiantes.length == 3) {
```